beacon-chain/blockchain/service.go

```diff
@@ -336,38 +334,20 @@ func (s *Service) Status() error {
 
 // This gets called when beacon chain is first initialized to save genesis data (state, block, and more) in db.
 func (s *Service) saveGenesisData(ctx context.Context, genesisState iface.BeaconState) error {
-	stateRoot, err := genesisState.HashTreeRoot(ctx)
-	if err != nil {
-		return err
+	if err := s.cfg.BeaconDB.SaveGenesisData(ctx, genesisState); err != nil {
+		return errors.Wrap(err, "could not save genesis data")
+	}
+	genesisBlk, err := s.cfg.BeaconDB.GenesisBlock(ctx)
+	if err != nil || genesisBlk == nil {
+		return errors.Wrap(err, "could not load genesis block")
```

Comment:
This would just return `nil` if `nil || true`

```go
var err error
require.Equal(t, nil, errors.Wrap(err, "could not load genesis block"))
```

Perhaps it's missing a test case? 🤔

Reply:
Yes, that is fine. Basically would be an error like "could not load genesis block: nil" which would be indicative that it was a nil genesis block. 

I suppose I could expand this if anyone didnt understand that it was nil

Reply:
I dont think it would be `could not load genesis block: nil`, the err would literally be `nil`

Reply:
Got it, i'll use fmt.Errorf, ty